src/Adapter/Country/ValueObject/Country.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Country\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Country\Exception\CountryConstraintException;
+
+class Country
+{
+    /** @var int */
+    private $id;
+    /** @var string */
+    private $name;
+
+    /**
+     * Country constructor.
+     *
+     * @param int $id
+     * @param string $name
+     */
+    public function __construct(int $id, string $name)
+    {
+        $this->assertIsIntegerGreaterThanZero($id);
+        $this->id = $id;
+        $this->name = $name;
+    }
+
+    /**
+     * Validates that the value is integer and is greater than zero
+     *
+     * @param $value
+     *
+     * @throws CountryConstraintException
+     */
+    private function assertIsIntegerGreaterThanZero($value)
```

Comment:
type value as int and as we have types, we dont need to assert if its int anymore, just 0.